cmd/vcluster/cmd/snapshot/create.go

```diff
@@ -0,0 +1,62 @@
+package snapshot
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/loft-sh/vcluster/pkg/config"
+	"github.com/loft-sh/vcluster/pkg/constants"
+	setupconfig "github.com/loft-sh/vcluster/pkg/setup/config"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/loft-sh/vcluster/pkg/snapshot"
+	"github.com/spf13/cobra"
+)
+
+func NewCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "create vCluster snapshots",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			// parse vCluster config
+			vConfig, err := config.ParseConfig(constants.DefaultVClusterConfigLocation, os.Getenv("VCLUSTER_NAME"), nil)
+			if err != nil {
+				return err
+			}
+
+			envOptions, err := snapshot.ParseOptionsFromEnv()
+			if err != nil {
+				return fmt.Errorf("failed to parse options from environment: %w", err)
+			}
+
+			restClient, vClusterNamespace, err := setupconfig.InitClientConfig()
+			if err != nil {
+				return err
+			}
+
+			kubeClient, err := kubernetes.NewForConfig(restClient)
+			if err != nil {
+				return err
+			}
+
+			requet, err := snapshot.CreateSnapshotRequestResources(cmd.Context(), vConfig.Name, vClusterNamespace, envOptions, kubeClient)
```

Comment:
parameters order is different, vCluster namespace goes before name

```suggestion
			requet, err := snapshot.CreateSnapshotRequestResources(cmd.Context(), vClusterNamespace, vConfig.Name, envOptions, kubeClient)
```

Reply:
Did you maybe check if `vCluster.Name` is set here to a non-empty value? IIRC I tried using the name (or namespace) from the config (in the CLI code though), but it was empty.

Reply:
Yes, it gets vCluster name from the `VCLUSTER_NAME` env var

```
Name:myvcluster2 HostNamespace: Path:/var/lib/vcluster/config.yaml
```